scene/main/viewport.cpp

```diff
@@ -1448,7 +1448,7 @@ void Viewport::_gui_show_tooltip() {
 			&tooltip_owner);
 	gui.tooltip_text = gui.tooltip_text.strip_edges();
 
-	if (gui.tooltip_text.is_empty()) {
+	if (gui.tooltip_text.is_empty() && !(tooltip_owner && tooltip_owner->is_force_show_tooltip())) {
```

Comment:
Is a new virtual method really necessary? We can check if `make_custom_tooltip` return non `null` control and assume it should be displayed. It's called a few lines below anyway, so some code can be rearranged without extra calls:

```c++
...
gui.tooltip_text = gui.tooltip_text.strip_edges();

if (!tooltip_owner) {
	return;
}

// Controls can implement `make_custom_tooltip` to provide their own tooltip.
// This should be a Control node which will be added as child to a TooltipPanel.
Control *base_tooltip = tooltip_owner->make_custom_tooltip(gui.tooltip_text);

if (gui.tooltip_text.is_empty() && !base_tooltip) {
	return; // Nothing to show.
}

// Remove previous popup if we change something.
...
```